src/components/Buttons/Copy/Copy.tsx

```diff
@@ -0,0 +1,80 @@
+import { useState } from 'react';
+import cn from 'classnames';
+import { Button, ButtonSize, ButtonVariant, IconCatalog, Tooltip } from '~/components';
+
+interface CopyBtnProps {
+  /**
+   * Specify the class name for the button
+   */
+  className?: string;
+  /**
+   * Specify the target to be copied
+   */
+  target: HTMLElement;
+  /**
+   * Specify the size of the button
+   */
+  size?: ButtonSize;
+  /**
+   * Specify whether to invert the button colors
+   */
+  invert?: boolean;
+  /**
+   * Specify the variant of the button
+   */
+  variant?: ButtonVariant;
+  /**
+   * Specify whether the icon should be solid
+   */
+  isIconSolid?: boolean;
+  /**
+   * Specify whether the button is disabled
+   */
+  isDisabled?: boolean;
+}
+
+export const CopyBtn = ({
+  className,
+  target,
+  size = ButtonSize.base,
+  invert = true,
+  variant = ButtonVariant.ghost,
+  isIconSolid = false,
+  isDisabled = false,
+}: CopyBtnProps) => {
```

Comment:
```suggestion
export const CopyBtn = ({
  className,
  target,
  size = ButtonSize.base,
  invert = true,
  variant = ButtonVariant.ghost,
  isIconSolid = false,
  isDisabled = false,
  ...restOfProps
}: CopyBtnProps) => {
```

Este ejemplo no vas a encontrar en TextInput... ahi te muestra como extende de Input, y hace el **...restOfProps**

Si no entiendes algo me escribes